spec/currency_spec.rb

```diff
@@ -0,0 +1,81 @@
+require File.expand_path(File.dirname(__FILE__) + '/spec_helper')
+
+describe "Currency" do
+  let(:currency) { Money::Currency.new('usd') }
+  let(:currency_data) {{
+    "iso_code": "USD",
+    "name": "United States Dollar",
+    "subunit_to_unit": 100,
+    "iso_numeric": "840",
+    "smallest_denomination": 1,
+    "minor_units": 2
+  }}
+
+  describe "#new" do
+    it "is constructable with a uppercase string" do
+      expect(Money::Currency.new('USD').iso_code).to eq('USD')
+    end
+
+    it "is constructable with a symbol" do
+      expect(Money::Currency.new(:usd).iso_code).to eq('USD')
+    end
+
+    it "is constructable with a lowercase string" do
+      expect(Money::Currency.new('usd').iso_code).to eq('USD')
+    end
+
+    it "has currency data accessible" do
+      currency_data.keys.each do |attribute|
+        expect(currency.public_send(attribute)).to eq(currency_data[attribute])
+      end
+    end
+
+    it "raises if the currency is invalid" do
```

Comment:
for new valid is split in 3 tests above:
`it "is constructable with a uppercase string" do` , `it "is constructable with a symbol" do`, `it "is constructable with a lowercase string" do`